cmd/vsphere-xcopy-volume-populator/certificate-tool/Makefile

```diff
@@ -0,0 +1,98 @@
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
+CONFIG_FILE ?= assets/config/static_values.yaml
+ENV_FILE ?= assets/config/.env
+-include $(ENV_FILE)
+
+SHELL := /usr/bin/env bash
+
+TEST_NAMESPACE = $(shell yq e '.test-namespace' $(CONFIG_FILE))
+POD_NAMESPACE = $(shell yq e '.pod-namespace' $(CONFIG_FILE))
+STORAGE_VENDOR   = $(shell yq e '.storage-vendor' $(CONFIG_FILE))
+KUBECONFIG       = $(shell yq e '.kubeconfig' $(CONFIG_FILE))
+TEST_IMAGE_LABEL = $(shell yq e '.test-image-label' $(CONFIG_FILE))
+TEST_LABELS      = $(shell yq e '.test-labels' $(CONFIG_FILE))
+TEST_POPULATOR_IMAGE = $(shell yq e '.test-populator-image' $(CONFIG_FILE))
+CONTROLLER_PATH  = $(shell yq e '.controller-path' $(CONFIG_FILE))
+
+REQUIRED_VARS := STORAGE_USER STORAGE_PASSWORD STORAGE_URL VSPHERE_USER VSPHERE_PASSWORD VSPHERE_URL
+
+$(foreach var,$(REQUIRED_VARS),\
+  $(if $(value $(var)),,$(error Environment variable $(var) is required)))
+
+
+# TODO: pin the exact commit
+OPENSTACK_POPULATOR ?= https://raw.githubusercontent.com/kubev2v/forklift/main/operator/config/crd/bases/forklift.konveyor.io_ovirtvolumepopulators.yaml
+OVIRT_POPULATOR     ?= https://raw.githubusercontent.com/kubev2v/forklift/main/operator/config/crd/bases/forklift.konveyor.io_openstackvolumepopulators.yaml
+VSPHERE_POPULATOR   ?= https://raw.githubusercontent.com/kubev2v/forklift/main/operator/config/crd/bases/forklift.konveyor.io_vspherexcopyvolumepopulators.yaml
+CLUSTER_ROLE   ?= https://raw.githubusercontent.com/kubev2v/forklift/main/operator/config/rbac/populator-controller/role.yaml
+ROLE_BINDING   ?= https://raw.githubusercontent.com/kubev2v/forklift/main/operator/config/rbac/populator-controller/role_binding.yaml
+SERVICE_ACCOUNT   ?= https://raw.githubusercontent.com/kubev2v/forklift/main/operator/config/rbac/populator-controller/service_account.yaml
+VM_NAME      ?= certificate-vm
+
+BINARY = certificate-tool
+
+.PHONY: help build prepare test-xcopy all clean
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} \
+		/^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } \
+		/^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+check-tools:
+	@command -v yq >/dev/null || (echo "yq not installed" && exit 1)
+	@command -v kubectl >/dev/null || (echo "kubectl not installed" && exit 1)
+	@command -v go >/dev/null || (echo "go not installed" && exit 1)
+
+build: ## Build the certificate-tool CLI binary.
+	go build -o $(BINARY) .
+
+prepare: check-tools build ## Run certificate-tool create-env subcommand.
+	@kubectl apply -f $(OPENSTACK_POPULATOR)
```

Comment:
apply is mutating the file you are working on - it sets an annotation on the resource in question with the former state. I think in most cases we need `create` or `replace`

Reply:
I think we can get rid of kubectl completely, and use the client-go library. It supports KUBECONFIG anyway, and with that we loose the dependency on kubectl as a tool, and we also just read `config/static_values.yaml` into the code and then replace the values in the go code itself, and remove the stage where we read yamls to env vars and so on... so we won't need `yq` tool either, because we can just read yaml using go. So we would be left only with go code. which means that people can run the binary , with no deps at all